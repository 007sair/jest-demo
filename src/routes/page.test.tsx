import App from '@/routes/page';
import { render, screen } from '@testing-library/react';

describe('Page.tsx', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a heading', () => {
    render(<App />);

    const heading = screen.getByRole('heading', { name: /demo/i });
    // or
    // const heading = screen.getByRole('heading', { level: 2 });

    expect(heading).toBeInTheDocument();
  });
});
