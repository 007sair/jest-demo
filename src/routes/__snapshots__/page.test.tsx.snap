// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page.tsx should take a snapshot 1`] = `
<DocumentFragment>
  <h2>
    Demo
  </h2>
  <ul>
    <li>
      <a
        data-testid="roter-link"
        href="/act"
      >
        /act
      </a>
    </li>
    <li>
      <a
        data-testid="roter-link"
        href="/global-style"
      >
        /global-style
      </a>
    </li>
    <li>
      <a
        data-testid="roter-link"
        href="/wait-for"
      >
        /wait-for
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
